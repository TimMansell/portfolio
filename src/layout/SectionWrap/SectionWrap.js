import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './SectionWrap.module.scss';

const SectionWrap = ({ container, type, id, children }) => {
  const sectionClasses = classnames(styles.section, {
    [styles.bgPrimary]: type === 'primary',
    [styles.bgSecondary]: type === 'secondary',
    [styles.bgTertiary]: type === 'tertiary',
  });

  const containerClasses = classnames(styles.container, {
    [styles.containerMedium]: container === 'medium',
    [styles.containerLarge]: container === 'large',
  });

  return (
    <section id={id} className={sectionClasses}>
      <div className={containerClasses}>{children}</div>
    </section>
  );
};

SectionWrap.propTypes = {
  container: PropTypes.oneOf(['medium', 'large']),
  type: PropTypes.oneOf(['primary', 'secondary', 'tertiary']).isRequired,
  id: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
};

export default SectionWrap;
