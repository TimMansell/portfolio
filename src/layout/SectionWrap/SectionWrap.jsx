import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './SectionWrap.module.scss';

const SectionWrap = ({ container, type, id, children }) => {
  const sectionClasses = classnames(styles.section, {
    [styles.bgPrimary]: type === 'primary',
    [styles.bgSecondary]: type === 'secondary',
    [styles.bgTertiary]: type === 'tertiary',
  });

  const containerClasses = classnames(styles.container, {
    [styles.containerMedium]: container === 'medium',
    [styles.containerLarge]: container === 'large',
    [styles.containerSmall]: container === 'small',
  });

  return (
    <section id={id} className={sectionClasses}>
      <div className={containerClasses}>{children}</div>
    </section>
  );
};

SectionWrap.propTypes = {
  container: PropTypes.oneOf(['medium', 'large', 'small']),
  type: PropTypes.oneOf(['primary', 'secondary', 'tertiary']).isRequired,
  id: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
};

export default SectionWrap;
