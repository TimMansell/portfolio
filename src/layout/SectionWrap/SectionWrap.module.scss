@import 'variables';

.section {
  padding-top: 80px;
  padding-bottom: 80px;
}

.container {
  width: 100%;
  padding-left: 20px;
  padding-right: 20px;
  margin-right: auto;
  margin-left: auto;

  @media (min-width: $bp-sm) {
    max-width: 540px;
  }

  @media (min-width: $bp-md) {
    max-width: 720px;
  }

  @media (min-width: $bp-lg) {
    max-width: 960px;
  }

  @media (min-width: $bp-xl) {
    max-width: 1140px;
  }
}

.medium {
  width: 85%;
  max-width: 85%;
}

.large {
  width: 100%;
  max-width: 100%;
}

.bgPrimary {
  composes: primary from '~scss/helpers/_backgrounds.scss';
}

.bgSecondary {
  composes: secondary from '~scss/helpers/_backgrounds.scss';
}

.bgTertiary {
  composes: tertiary from '~scss/helpers/_backgrounds.scss';
}
