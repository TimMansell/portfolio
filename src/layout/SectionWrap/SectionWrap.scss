@import 'utilities/breakpoints';

.section-wrap {
	position: relative;
	padding-top: 80px;
	padding-bottom: 80px;
}

.container {
	width: 100%;
	padding-left: 20px;
    padding-right: 20px;
	margin-right: auto;
	margin-left: auto; 

	@include media-breakpoint-up(sm){
		max-width: 540px;
	}

	@include media-breakpoint-up(md){
		max-width: 720px;
	}

	@include media-breakpoint-up(lg){
		max-width: 960px;
	}

	@include media-breakpoint-up(xl){
		max-width: 1140px; 
    }

    &--medium {
        width: 85%;
        max-width: 85%;
    }
    
    &--large {
        width: 100%;
        max-width: 100%;
    }
}