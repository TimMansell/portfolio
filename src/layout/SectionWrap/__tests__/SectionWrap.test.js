import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import SectionWrap from '../SectionWrap';

describe('Section', () => {
  it('should render my component', () => {
    // eslint-disable-next-line
    const wrapper = shallow(<SectionWrap />);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<SectionWrap/>).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it('should render a default component', () => {
    const wrapper = shallow(<SectionWrap />);

    expect(wrapper.hasClass('section-wrap')).toBeTruthy();
    expect(wrapper.find('.container')).toHaveLength(1);
  });

  it('should wrap an element', () => {
    const wrapper = shallow(<SectionWrap><p>test</p></SectionWrap>);

    expect(wrapper.find('p').text()).toEqual('test');
  });

  it('should display a primary background', () => {
    const wrapper = shallow(<SectionWrap background="primary" />);

    expect(wrapper.hasClass('bg--primary')).toBeTruthy();
  });

  it('should display a secondary background', () => {
    const wrapper = shallow(<SectionWrap background="secondary" />);

    expect(wrapper.hasClass('bg--secondary')).toBeTruthy();
  });

  it('should display a tertiary background', () => {
    const wrapper = shallow(<SectionWrap background="tertiary" />);

    expect(wrapper.hasClass('bg--tertiary')).toBeTruthy();
  });

  it('should render a medium container', () => {
    const wrapper = shallow(<SectionWrap container="medium" />);

    expect(wrapper.find('.container--medium')).toHaveLength(1);
  });

  it('should render a large container', () => {
    const wrapper = shallow(<SectionWrap container="large" />);

    expect(wrapper.find('.container--large')).toHaveLength(1);
  });

  it('should bind an id', () => {
    const wrapper = shallow(<SectionWrap id="test" />);

    expect(wrapper.find('#test')).toHaveLength(1);
  });
});
