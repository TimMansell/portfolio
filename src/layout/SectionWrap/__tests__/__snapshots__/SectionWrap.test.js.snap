// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should match snapshot 1`] = `
<section
  className="section-wrap"
  id=""
>
  <div
    className="container"
  />
</section>
`;
